compiler/env/defines.h

```diff
@@ -136,6 +139,9 @@
 #elif defined(__aarch64__)
 #  define HOST_ARCH ARCH_ARM64
 # define TR_HOST_64BIT 1
+#elif defined(__riscv) && defined(RISCV64)
+#  define HOST_ARCH ARCH_RISCV64
+# define TR_HOST_64BIT 1
```

Comment:
Sorry for jumping on this little late. 

I think we should align this code with changes done in PR #3907. See https://github.com/eclipse/omr/pull/3907/files#diff-9cf243719c23dca420e7d19d55be39cc

There we have this:
```
# elif defined(__riscv)
#  define HOST_ARCH ARCH_RISCV
#  if __riscv_xlen == 32
#    define TR_HOST_32BIT 1
#  elif __riscv_xlen == 64
#    define TR_HOST_64BIT 1
#  else
#    error "defines.h: unknown word size"
#  endif
```

I'll do a pass on this and send a aligning patch later this week.

Reply:
That sounds good, thanks!

Reply:
I already changed to `__riscv_xlen `. In addition, I will still keep `ARCH_RISCV64` there
```
#elif (defined(__riscv) && (64 == __riscv_xlen))
#  define HOST_ARCH ARCH_RISCV64
# define TR_HOST_64BIT 1
```
as we don't think we will support 32bit for now.

Reply:
I must say that I would still prefer "my" version:
(i) RV32 is not supported now and maybe not interesting for you and maybe not for J9, but I can (for)see others interested in OMR on RV32. Dunno about any specific need as of now, though. 
(ii) Using `ARCH_RISCV` (as opposed to `ARCH_RISCV64`) is consistent with other archs already supported - see `ARCH_POWER` or `ARCH_X86` (note, that ARM and AArch64 are really a different achitectures, so using `ARCH_ARM64` makes sense). 

We may (should?) add an explicit `#error "RV32 not yet supported"` somewhere to make the compilation to fail when run on RV32 (for now)

Reply:
There are too many restrictions on 32bit (memory, OS support/specific to linux, etc), in which case we won't consider that for the moment unless there are strong motivation or business requirement.  I personally don't object to `ARCH_RISCV `(I am using it somewhere in OpenJ9) but there should be no conflict with ARCH_RISCV64 as ARCH_RISCV64 represents 2 things: `riscv `and `64bit `. so there is no need to check everywhere with both `__riscv  `and `__riscv_xlen` to determine whether it is RISCV 64bit or not.

Reply:
> so there is no need to check everywhere with both __riscv and __riscv_xlen or equivalent to determine whether it is RISCV 64bit or not.

I don't think this is much of an issue. In arch-specific code, you know it's RV so you check only `TR_HOST_64BIT`. And for the rest of TR code that is arch-specific (IIRC debug code, insn printers and so on), it's not a big deal. And again, aligned with other archs. 

Anyways, let's see what others think.

Reply:
@DanHeidinga , should we leave the space on 32 bit for the future use?  If so, we can change from `ARCH_RISCV64 `  to `ARCH_RISCV` everywhere.

Reply:
I think @janvrany is right, we should use `ARCH_RISCV`, but I'm sure @0xdaryl can tell us which way to go.

Reply:
Already discussed with @DanHeidinga,  we will change to `ARCH_RISCV ` to align with code being merged to the OMR but there is no intention to do 32bit on OpenJ9 in the fear future. To avoid any conflict when merging to OMR, only my changes involved will be modified accordingly.

Reply:
I've updated everywhere to support `ARCH_RISCV ` after veriying the cross-build on RISC-V.

Reply:
@janvrany ,  your change without the underscores has been added at `cmake/modules/platform/arch/riscv.cmake`.